src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -165,6 +176,95 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
     }
 
     /**
+     * Hold the pair of number of avaliable and needed StackItem
+     */
+    public static class BuildingBuilderResource
+    {
+        public enum RessourceAvailability
+        {
+            NOT_NEEDED,
+            DONT_HAVE,
+            NEED_MORE,
+            HAVE_ENOUGHT
+        }
+
+        private String name;
+        private ItemStack itemStack;
+        private int amountAvailable;
+        private int amountNeeded;
+        private int amountPlayer;
+
+        BuildingBuilderResource(final String name, final ItemStack itemStack, final int amountAvailable, final int amountNeeded)
+        {
+            this.name = name;
+            this.itemStack = itemStack;
+            this.amountAvailable = amountAvailable;
+            this.amountNeeded = amountNeeded;
+            this.amountPlayer = 0;
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public ItemStack getItemStack()
+        {
+            return itemStack;
+        }
+
+
+        public int getAvailable()
+        {
+            return amountAvailable;
+        }
+
+        public void setAvailable(final int amount)
+        {
+            amountAvailable=amount;
+        }
+
+        public int getNeeded()
+        {
+            return amountNeeded;
+        }
+
+        public void setPlayerAmount(final int amount)
+        {
+            amountPlayer = amount;
+        }
+
+        public int getPlayerAmount()
+        {
+            return amountPlayer;
+        }
+
+        public RessourceAvailability getAvailabilityStatus()
+        {
+            if (amountNeeded >amountAvailable)
+            {
+                if (amountPlayer==0)
+                {
+                    return RessourceAvailability.DONT_HAVE;
+                }
+                if (amountPlayer<(amountNeeded-amountAvailable))
+                {
+                    return RessourceAvailability.NEED_MORE;
+                }
+                return RessourceAvailability.HAVE_ENOUGHT;
+            }
+            return RessourceAvailability.NOT_NEEDED;
+        }
+
+        public String toString()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)